documents/Specification/MaterialX.Proposals.md

```diff
@@ -0,0 +1,342 @@
+
+<a id="node-curveadjust"> </a>
+
+* **`curveadjust`** (NG): output a smooth remapping of input values using the centripetal Catmull-Rom cubic spline curve defined by specified knot values, using an inverse spline lookup on input knot values and a forward spline through output knot values.  All channels of the input will be remapped using the same curve.
+    * `in` (float or colorN or vectorN): the input value or nodename
+    * `numknots` (uniform integer): the number of values in the knots and knotvalues arrays
+    * `knots` (uniform floatarray): the list of input values defining the curve for the remapping.  At least 2 and at most 16 values must be provided.
+    * `knotvalues` (uniform floatarray): the list of output values defining the curve for the remapping.  Must be the same length as knots.
+
+<a id="node-curvelookup"> </a>
+
+* **`curvelookup`** (NG): output a float, colorN or vectorN value smoothly interpolated between a number of knotvalue values, using the position of in within knots as the knotvalues interpolant.
+    * `in` (float): the input interpolant value or nodename
+    * `numknots` (uniform integer): the number of values in the knots and knotvalues arrays
+    * `knots` (uniform floatarray): the list of knot values to interpolate in within.  At least 2 and at most 16 values must be provided.
+    * `knotvalues` (uniform floatarray or colorNarray or vectorNarray): the values at each knot position to interpolate between. Must be the same length as knots.
+
+
+
+### Compositing Nodes
+
+
+
+### Conditional Nodes
+
+<a id="node-ifelse"> </a>
+
+* **`ifelse`**: output the value of one of two input streams, according to whether the value of a boolean selector input is true or false
+    * `infalse`, `intrue` (float or color<em>N</em> or vector<em>N</em>): the values or nodenames to select from based on the value of the `which` input.  The types of the various `in<em>N</em>` inputs must match the type of the `switch` node itself.  The default value of all `in<em>N</em>` inputs is 0.0 in all channels.
+    * `which` (boolean): a selector to choose which input to take values from; default is "false".
+
+
+
+### Channel Nodes
+
+<a id="node-extractrowvector"> </a>
+
+* **`extractrowvector`**: extract the specified row vector number from a matrix<em>N</em> stream.
+    * `in` (matrix<em>N</em>): the input value or nodename
+    * `index` (integer): the row number to extract, should be 0-2 for matrix33 streams, or 0-3 for matrix44 streams.
+
+<a id="node-separatecolor4"> </a>
+
+* **`separatecolor4`** (NG): output the RGB and alpha channels of a color4 as separate outputs.
+    * `in` (color4): the input value or nodename
+    * `outcolor` (output, color3): the RGB channel values.
+    * `outa` (output, float): the value of the alpha channel.
+
+
+
+### Convolution Nodes
+
+<a id="node-blur"> </a>
```

Comment:
Although our reference implementation of the `blur` node in MaterialX is not yet up to the quality level of the other MaterialX nodes, it does have initial implementations in GLSL/OSL, and these implementations are validated in our test suite.

Given that context, it may make more sense to leave this node in the official specification, and to clarify in the README of our data libraries which nodes (including `blur`) have implementations that are believed to require additional work to reach the level of quality we expect.

Reply:
Good point, change made.